### PR TITLE
Deduplicate implementation of get_first_{bytes|five} functions

### PR DESCRIPTION
**Pull request by dralley**

### CODE REVIEW

File: src/utils.rs
Comment (dralley) [inline src/utils.rs:10]: Instead of requiring a concrete type you can just verify that the trait bounds are satisfied - then this no longer needs to be copied between the modules so long as no features from the additional trait bounds are used. `Box<Read + Send + whatever + 'a>` still implements `Read` so it works fine.  And there's no specialization for the versions with stricter trait bounds.

**Review by luizirber — APPROVED** [review]
Awesome! I'll let @natir take a look too before merging, and I wonder if we can use the same approach to limit the proliferation of submodules for different generic bounds

**Pull request merged**
